src/components/profiles/Profile.js

```diff
@@ -21,20 +21,23 @@ export class Profile extends Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    if (nextProps.getProfileInitiated === true) {
+    if (nextProps.profileRetrievalInitiated === true) {
       this.setState({ loaded: true });
```

Comment:
Don't check for the loaded value from here, just set it to the component props and read it from there in the render method.